internal/bminventory/inventory.go

```diff
@@ -290,8 +290,17 @@ func (b *bareMetalInventory) updatePullSecret(pullSecret string, log logrus.Fiel
 	return pullSecret, nil
 }
 
-func (b *bareMetalInventory) setDefaultRegisterClusterParams(ctx context.Context, params installer.V2RegisterClusterParams) (installer.V2RegisterClusterParams, error) {
+func (b *bareMetalInventory) setDefaultRegisterClusterParams(ctx context.Context, params installer.V2RegisterClusterParams, id strfmt.UUID) (installer.V2RegisterClusterParams, error) {
 	log := logutil.FromContext(ctx, b.log)
+
+	if params.NewClusterParams.APIVip != "" && len(params.NewClusterParams.APIVips) == 0 {
+		params.NewClusterParams.APIVips = []*models.APIVip{{IP: models.IP(params.NewClusterParams.APIVip), ClusterID: id}}
+	}
+
+	if params.NewClusterParams.IngressVip != "" && len(params.NewClusterParams.IngressVip) == 0 {
```

Comment:
What happens of both the singular and plural variables have values ?  Will there be a conflict ?

Reply:
yes. this is also being tested both in unit and subsystem

Reply:
see `handleApiVipBackwardsCompatibility` function in this PR

Reply:
Didn't find it

Reply:
https://github.com/nmagnezi/assisted-service/blob/bd0906f5729b8ef7ce75ff3289e485c4b722f8c8/internal/cluster/validations/validations.go#L318-L332